app-ios/tutanota/Sources/Alarms/AlarmModel.swift

```diff
@@ -176,6 +177,13 @@ struct LazyEventSequence : Sequence, IteratorProtocol {
         occurenceDate: occurrenceDate
       )
       ocurrenceNumber += 1
+      
+      while exclusionNumber < repeatRule.excludedDates.count && repeatRule.excludedDates[exclusionNumber] < occurrenceDate {
+        exclusionNumber += 1
+      }
+      if (exclusionNumber < repeatRule.excludedDates.count && repeatRule.excludedDates[exclusionNumber] == occurrenceDate) {
+        return self.next()
```

Comment:
I hope this recursion won't bite us in the butt but I also see no other easy way